pkg/utils/k8sutil.go

```diff
@@ -421,6 +370,58 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 		},
 	}
 
+	if len(funcObj.Spec.Template.Spec.Containers) == 0 {
```

Comment:
Move this to after the deepcopy and modify the copied (deployment) version.  The goal is to avoid modifying the funcObject passed to us, because it might be a pointer into a shared informer cache (for example).

If we were using another programming language, we would have used a `const` keyword to protect the passed funcObj argument, but with golang we just have to be vigilant :(